File: examples/git.java
import com.fizzed.blaze.Contexts;
import org.eclipse.jgit.lib.Repository;
import org.eclipse.jgit.storage.file.FileRepositoryBuilder;
import org.eclipse.jgit.api.Git;
import org.eclipse.jgit.api.Status;
import org.slf4j.Logger;

public class git {
    static final private Logger log = Contexts.logger();

    public void status() throws Exception {
        Repository repo = new FileRepositoryBuilder()
            .readEnvironment()  // scan environment GIT_* variables
            .findGitDir()       // scan up the file system tree
            .build();
        
        try (Git git = new Git(repo)) {
            Status status = git.status().call();
            log.info("Added: " + status.getAdded());
            log.info("Changed: " + status.getChanged());
            log.info("Conflicting: " + status.getConflicting());
            log.info("ConflictingStageState: " + status.getConflictingStageState());
            log.info("IgnoredNotInIndex: " + status.getIgnoredNotInIndex());
            log.info("Missing: " + status.getMissing());
            log.info("Modified: " + status.getModified());
            log.info("Removed: " + status.getRemoved());
            log.info("Untracked: " + status.getUntracked());
            log.info("UntrackedFolders: " + status.getUntrackedFolders());
        }
    }
    
}

File: examples/git.js
/* global Packages, Contexts */

var Imports = JavaImporter(
    Packages.com.fizzed.blaze.Contexts,
    Packages.org.eclipse.jgit.lib.Repository,
    Packages.org.eclipse.jgit.storage.file.FileRepositoryBuilder,
    Packages.org.eclipse.jgit.api.Git,
    Packages.org.eclipse.jgit.api.LogCommand,
    Packages.org.eclipse.jgit.api.errors.GitAPIException,
    Packages.org.eclipse.jgit.lib.Ref,
    Packages.org.eclipse.jgit.revwalk.RevCommit);

with (Imports) {

    var log = Contexts.logger();

    var status = function() {
        repo = new FileRepositoryBuilder()
            .readEnvironment()  // scan environment GIT_* variables
            .findGitDir()       // scan up the file system tree
            .build();

        g = new Git(repo);

        status = g.status().call();
        log.info("Added: {}", status.getAdded());
        log.info("Changed: {}", status.getChanged());
        log.info("Conflicting: {}", status.getConflicting());
        log.info("ConflictingStageState: {}", status.getConflictingStageState());
        log.info("IgnoredNotInIndex: {}", status.getIgnoredNotInIndex());
        log.info("Missing: {}", status.getMissing());
        log.info("Modified: {}", status.getModified());
        log.info("Removed: {}", status.getRemoved());
        log.info("Untracked: {}", status.getUntracked());
        log.info("UntrackedFolders: {}", status.getUntrackedFolders());
    };
    
}
